src/BlockTypes/ProductCategoryList.php

```diff
@@ -48,12 +48,57 @@ protected function get_block_type_supports() {
 	}
 
 	/**
-	 * Register script and style assets for the block type before it is registered.
+	 * Overwrite parent method to prevent script registration.
 	 *
-	 * This registers the scripts; it does not enqueue them.
+	 * It is necessary to register and enqueues assets during the render
+	 * phase because we want to load assets only if the block has the content.
 	 */
 	protected function register_block_type_assets() {
-		parent::register_block_type_assets();
-		$this->register_chunk_translations( [ $this->block_name ] );
+		return null;
+	}
+
+	/**
+	 * Register the context.
+	 */
+	protected function get_block_type_uses_context() {
+		return [ 'query', 'queryId', 'postId' ];
+	}
+
+	/**
+	 * Include and render the block.
+	 *
+	 * @param array    $attributes Block attributes. Default empty array.
+	 * @param string   $content    Block content. Default empty string.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		if ( ! empty( $content ) ) {
+			parent::register_block_type_assets();
+			$this->register_chunk_translations( [ $this->block_name ] );
+			return $content;
+		}
+
+		$post_id                  = $block->context['postId'];
+		$product                  = wc_get_product( $post_id );
+		$product_categories_terms = get_the_terms( $product->get_id(), 'product_cat' );
+
+		$output  = '';
+		$output .= '
+			<div class="wc-block-components-product-category-list">
+				' . __( 'Categories:', 'woo-gutenberg-products-block' )
+				. '<ul>';
+
+		foreach ( $product_categories_terms as $product_category_term ) {
+			$output .= '
+				<li class="category-list-item-' . $product_category_term->slug . '">
+					<a href="' . get_term_link( $product_category_term->term_id ) . '">' . $product_category_term->name . '</a>
```

Comment:
The problem with the spaces has been a little bit tricky. In the end, this is the fix:

```suggestion
					<a href="' . get_term_link( $product_category_term->term_id ) . '">' . $product_category_term->name . '</a></li>
```

I found different links that discuss this problem:
- https://stackoverflow.com/questions/5078239/how-to-remove-the-space-between-inline-inline-block-elements
- https://css-tricks.com/fighting-the-space-between-inline-block-elements/


![image](https://user-images.githubusercontent.com/4463174/201692360-1dcdb19e-2437-41b7-a110-ee095be4afb8.png)

Reply:
Ah, Got it. Thanks for sharing the link. The info mentioned [here](https://css-tricks.com/fighting-the-space-between-inline-block-elements/#aa-remove-the-spaces) is useful. 🙌🏻